### PR TITLE
add multichannel

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
Following on discussions #481 and with @bioimage-analysis this PR adds a convenience method for displaying multichannel image data. It also renames `multichannel` to `rgb` on the image layer. This PR still needs tests.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/components/viewer_model.py [hunk 1]
```diff
@@ -415,6 +415,117 @@ def add_image(
...
+        -------
+        layers : list of :class:`napari.layers.Image`
+            The newly-created image layers.
+        """
+        n_images = data.shape[channel]
+
+        base_colormaps = ['red', 'green', 'blue', 'cyan', 'yellow', 'magenta']
+        if colormap is None:
+            indices = np.arange(n_images) % len(base_colormaps)
+            expanded_colormaps = [base_colormaps[i] for i in indices]
```
Comment (jni) [inline napari/components/viewer_model.py]:
```python
expanded_colormaps = itertools.cycle(base_colormaps)
```

=)

File: napari/components/viewer_model.py [hunk 2]
```diff
@@ -415,6 +415,117 @@ def add_image(
...
+            Whether the layer visual is currently being displayed.
+
+        Returns
+        -------
+        layers : list of :class:`napari.layers.Image`
+            The newly-created image layers.
+        """
+        n_images = data.shape[channel]
+
+        base_colormaps = ['red', 'green', 'blue', 'cyan', 'yellow', 'magenta']
```
Comment (jni) [inline napari/components/viewer_model.py]: (a) I would put CYM before RGB, but
(b) before either, I might put BOP: https://twitter.com/christlet/status/1133726940127477761

Thoughts?

File: napari/components/viewer_model.py [hunk 3]
```diff
@@ -415,6 +415,117 @@ def add_image(
...
+        else:
+            expanded_colormaps = [colormap] * n_images
+
+        if contrast_limits is None:
+            expanded_contrast_limits = [None] * n_images
+        else:
+            if isinstance(contrast_limits, list):
+                expanded_contrast_limits = contrast_limits
+            else:
+                expanded_contrast_limits = [contrast_limits] * n_images
```
Comment (jni) [inline napari/components/viewer_model.py]: For all of these, I would say, (1) make sure it's iterable, then (2) use itertools.cycle on the iterable (regardless of length — could be an iterator with no `__len__`). I think that will make the code more readable.

File: napari/components/viewer_model.py [hunk 4]
```diff
@@ -415,6 +415,117 @@ def add_image(
...
+
+        if name is None:
+            expanded_names = [None] * n_images
+        elif isinstance(name, list):
+            expanded_names = name
+        else:
+            expanded_names = [name] * n_images
+
+        layers = []
+        for i in range(data.shape[channel]):
```
Comment (jni) [inline napari/components/viewer_model.py]: Then with cycle you'd need `for i, cmap, clims, name in zip(range(data.shape[channel]), colormaps, ...`. Though you could also use `itertools.islice`.

File: napari/layers/image/image.py [hunk 5]
```diff
@@ -24,9 +20,9 @@ class Image(Layer):
     ----------
     data : array
         Image data. Can be N dimensional. If the last dimension has length
-        3 or 4 can be interpreted as RGB or RGBA if multichannel is `True`.
-    multichannel : bool
-        Whether the image is multichannel RGB or RGBA if multichannel. If
+        3 or 4 can be interpreted as RGB or RGBA if rgb is `True`.
+    rgb : bool
+        Whether the image is rgb RGB or RGBA if rgb. If
```
Comment (jni) [inline napari/layers/image/image.py]: "if rgb" at the end can be removed...

File: napari/layers/image/image.py [hunk 6]
```diff
@@ -66,26 +62,26 @@ class Image(Layer):
...
     data : array
         Image data. Can be N dimensional. If the last dimension has length 3
-        or 4 can be interpreted as RGB or RGBA if multichannel is `True`.
+        or 4 can be interpreted as RGB or RGBA if rgb is `True`.
     metadata : dict
         Image metadata.
-    multichannel : bool
-        Whether the image is multichannel RGB or RGBA if multichannel. If not
+    rgb : bool
+        Whether the image is rgb RGB or RGBA if rgb. If not
```
Comment (jni) [inline napari/layers/image/image.py:68]: as above

**Review by jni — APPROVED** [review]
Love it! Left some minor comments, and of course tests would be nice! =)

File: napari/components/viewer_model.py (hunk 1 above)
Comment (sofroniewn) [reply napari/components/viewer_model.py]: done, thanks!

File: napari/components/viewer_model.py (hunk 2 above)
Comment (sofroniewn) [reply napari/components/viewer_model.py]: For now I'll switch CYM with RGB - BOP looks great, but we don't have `orange` and `purple` as defaults. We should add those first and then we can revisit this

File: napari/layers/image/image.py (hunk 5 above)
Comment (sofroniewn) [reply napari/layers/image/image.py]: fixed

File: napari/layers/image/image.py (hunk 6 above)
Comment (sofroniewn) [reply napari/layers/image/image.py:68]: fixed

File: napari/components/viewer_model.py (hunk 4 above)
Comment (sofroniewn) [reply napari/components/viewer_model.py]: done

File: napari/components/viewer_model.py (hunk 3 above)
Comment (sofroniewn) [reply napari/components/viewer_model.py]: done

**Pull request merged**